docs/design/002-containerized-gadgets.md

```diff
@@ -0,0 +1,226 @@
+the output from BPF_SEQ_PRINTF() calls in the following format:
+```
+[event]
+fieldName1=fieldValue1\0
+fieldName2=fieldValue2\0
+```
+
+Note that some fields such as `comm` can contain a new line character ("\n"), so we can't use only that as a separator.
+
+Alternatively, the bpf module can contain a global variable with the same name as the iterator plus a suffix "_output".
+So the BTF information can be attached to that variable.
+```
+const volatile event_t myiterator_output = {};
+
+SEC("iter/task_file")
+int myiterator(struct bpf_iter__task_file *ctx)
+```
+
+Problem: if that variable is not used by any program, the cilium/ebpf loader will complain.
+
+## Toppers
+
+TODO
+
+# Reimplementing existing gadgets as containerized gadgets
+
+For each existing gadget, we list the additional feature needed that are not yet supported for containerized containers
+in Inspektor Gadget. For this exercise, we look at each gadget tracer.go and *.bpf.c file.
+
+| Gadget                 | Feature needed                                                                  | Difficulty |
+|------------------------|---------------------------------------------------------------------------------|------------|
+| advise network-policy  |                                                                                 |            |
+| advise seccomp-profile |                                                                                 |            |
+| audit seccomp          |                                                                                 |            |
+| profile block-io       |                                                                                 |            |
+| profile cpu            |                                                                                 |            |
+| profile tcprtt         |                                                                                 |            |
+| snapshot process       |                                                                                 |            |
+| snapshot socket        |                                                                                 |            |
+| top block-io           |                                                                                 |            |
+| top ebpf               |                                                                                 |            |
+| top file               |                                                                                 |            |
+| top tcp                |                                                                                 |            |
+| trace bind             | field_filter, bitfield_column, enum_convert, ifindex                            |            |
+| trace capabilities     |                                                                                 |            |
```

Comment:
- `enum_convert`
- int -> syscall name: Can we implement this as enum_convert given that syscalls are different per architecture?